extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/HttpUpgradePolicy.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.websockets.next;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A policy that controls which requests are allowed to upgrade the HTTP connection to a WebSocket connection.
+ * CDI beans implementing this interface are invoked on every request.
+ */
+public interface HttpUpgradePolicy {
```

Comment:
Hm, is "policy" a good name for this interface? I mean I don't have a better one, just thinking...

Reply:
Hm, the logic happens before the HTTP upgrade so I was thinking about `HttpUpgradeInitializer` or even `HttpUpgradeCheck`.... but these are far from being ideal.

Reply:
for `HttpUpgradeCheck` it just brings trouble that I don't know how to call method :-) `HttpUpgradeCheck#checkUpgrade` is not good.

I think all 3 would do for me. @mkouba if you want a time to think about it, let's take it. Otherwise please pick one as I don't have a hard opinion. Thanks.

Reply:
`HttpUpgradeCheck#perform()`? 

> if you want a time to think about it, let's take it. Otherwise please pick one as I don't have a hard opinion. Thanks.

Let's think about this for a while and we'll see... :-)

Reply:
@cescoffier @sberyozkin @geoand WDYM? I know that it's just a name, but still...

Reply:
Hi @michalvavrik @michalvavrik I see that Michal modeled it similarly to `HttpSecurityPolicy` which is about allowing or denying an access. I don't really mind,  `HttpUpgradeCheck#perform` works too, or may be also `HttpUpgradeController#check`